File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter04/Sponsor/MathClient/CustomerClientSponsor.vb
Imports System.Runtime.Remoting.Lifetime 'ISponsor and ILease

Public Class CustomerClientSponsor
   Inherits MarshalByRefObject
   Implements ISponsor
   
   Private mRenewCount As Integer = 0
      
   'Implements ISponsor.Renewal
   Public Function Renewal(leaseInfo As ILease) As TimeSpan _
      Implements ISponsor.Renewal
      
      If mRenewCount < 3 Then        
         mRenewCount += 1

         'Just a trace messsage for testing.
         Console.WriteLine("CustomerClientSponsor.Renewal()")
         Return TimeSpan.FromSeconds(5)       
      Else
         'Just a trace messsage for testing.
         Console.WriteLine("CustomerClientSponsor.Renewal(), returning 0 time!")
         Return TimeSpan.FromSeconds(0)     
      End If
   End Function
   
End Class

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter04/Sponsor/MathClient/CustomerClientSponsor.cs
using System;
using System.Runtime.Remoting.Lifetime;

namespace MathClient
{
   class CustomerClientSponsor : MarshalByRefObject, ISponsor
   {
      private int mRenewCount = 0;
      
      // Implements ISponsor.Renewal
      public TimeSpan Renewal(ILease leaseInfo)
      {
         if (mRenewCount < 3)
         {
            mRenewCount++;

            // Just a trace messsage for testing.
            Console.WriteLine("CustomerClientSponsor.Renewal()");
            return TimeSpan.FromSeconds(5);
         } 
         else 
         {
            // Just a trace messsage for testing.
            Console.WriteLine("CustomerClientSponsor.Renewal(), returning 0 time!");
            return TimeSpan.FromSeconds(0);
         }
      }
   }
}
